src/core/manager.c

```diff
@@ -1983,7 +1983,7 @@ int manager_startup(Manager *m, FILE *serialization, FDSet *fds, const char *roo
         /* If we are running in test mode, we still want to run the generators,
          * but we should not touch the real generator directories. */
         r = lookup_paths_init_or_warn(&m->lookup_paths, m->runtime_scope,
-                                      MANAGER_IS_TEST_RUN(m) ? LOOKUP_PATHS_TEMPORARY_GENERATED : 0,
+                                      FLAGS_SET(m->test_run_flags, MANAGER_TEST_RUN_GENERATORS) ? LOOKUP_PATHS_TEMPORARY_GENERATED : 0,
```

Comment:
Maybe, also document that transient services would be discarded when `--generators` is specified?